features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Create/APIProduct/APIProductCreateWrapper.jsx

```diff
@@ -199,6 +199,57 @@ export default function ApiProductCreateWrapper(props) {
             .saveProduct(apiData)
             .then((apiProduct) => {
                 Alert.info('API Product created successfully');
+                const body = {
+                    description: 'Initial Revision',
+                };
+                newAPIProduct.createProductRevision(apiProduct.id, body)
+                    .then((api1) => {
+                        const revisionId = api1.body.id;
+                        Alert.info('API Revision created successfully');
+                        const envList = settings.environment.map((env) => env.name);
+                        const body1 = [];
+                        if (envList && envList.length > 0) {
+                            if (envList.includes('Default')) {
+                                body1.push({
+                                    name: 'Default',
+                                    displayOnDevportal: true,
+                                });
+                            } else {
+                                body1.push({
+                                    name: envList[0],
+                                    displayOnDevportal: true,
+                                });
+                            }
+                        }
+                        newAPIProduct.deployProductRevision(apiProduct.id, revisionId, body1)
+                            .then(() => {
+                                Alert.info('API Revision Deployed Successfully');
+                            })
+                            .catch((error) => {
+                                if (error.response) {
+                                    Alert.error(error.response.body.description);
+                                } else {
+                                    const message = 'Something went wrong while deploying the API Revision';
+                                    Alert.error(intl.formatMessage({
+                                        id: 'Apis.Create.Default.APICreateDefault.error.errorMessage.deploy.revision',
```

Comment:
Let's change this ID to reflect the file name

Reply:
Fixed